File: 2025-08-August-GFG-POTD/August 05 - Palindrome Sentence/Solution.java
class Solution {
    public boolean isPalinSent(String s) {
        // Step 1: Normalize the string
        StringBuilder sb = new StringBuilder();
        for (char c : s.toCharArray()) {
            if (Character.isLetterOrDigit(c)) {
                sb.append(Character.toLowerCase(c));
            }
        }
        
        // Step 2: Check if the cleaned string is a palindrome
        int left = 0;
        int right = sb.length() - 1;
        
        while (left < right) {
            if (sb.charAt(left) != sb.charAt(right)) {
                return false;
            }
            left++;
            right--;
        }
        
        return true;
    }
}

File: 2025-08-August-GFG-POTD/August 05 - Palindrome Sentence/Solution.cpp
class Solution {
public:
    bool isPalinSent(string s) {
        // Step 1: Normalize the string
        string cleaned;
        for (char c : s) {
            if (isalnum(c)) {
                cleaned += tolower(c);
            }
        }
        
        // Step 2: Check if the cleaned string is a palindrome
        int left = 0, right = cleaned.size() - 1;
        while (left < right) {
            if (cleaned[left] != cleaned[right]) {
                return false;
            }
            left++;
            right--;
        }
        
        return true;
    }
};
